hudi-common/src/main/java/org/apache/hudi/common/engine/TaskContextSupplier.java

```diff
@@ -35,4 +36,8 @@ public abstract class TaskContextSupplier implements Serializable {
   public abstract Supplier<Long> getAttemptIdSupplier();
 
   public abstract Option<String> getProperty(EngineProperty prop);
+
+  public Option<HoodieReaderContext> getReaderContext(HoodieTableMetaClient metaClient, boolean useReaderContext) {
```

Comment:
There is no usage for this method, and I don't think the task comtext supplier is the role to hold the reader context.

Reply:
Removed.  This was from the initial PoC code which was later removed.